src/schema/datetime.rs

```diff
@@ -25,6 +25,47 @@ use std::str::FromStr;
 
 const NANOSECONDS_IN_SECOND: u32 = 1_000_000_000;
 
+const MILLISECONDS_IN_SECOND: i64 = 1_000;
+
+const SECONDS_IN_DAY: i64 = 86_400;
+
+// Number of days between Apache Arrow / UNIX epoch (1970-01-01)
+// and PostgreSQL epoch (2000-01-01).
+const POSTGRES_BASE_DATE_OFFSET: Days = Days::new(10_957);
+
+/// Converts an [`i32`] stored in [`arrow::array::types::Date32Type`] to PostgresSQL TimestampWithTimeZone
+///
+/// Takes into account [`arrow::array::types::Date32Type`] stores the number of days
+/// elapsed since UNIX epoch (1970-01-01).
+/// Postgres [`datum::TimestampWithTimeZone`] type takes timestamp in microseconds
+/// with epoch (2000-01-01)
+#[inline(always)]
+pub(crate) fn arrow_date32_to_postgres_timestamps(
+    timestamp_in_days: i32,
+) -> Result<Option<TimestampWithTimeZone>, FromTimeError> {
+    arrow_date64_to_postgres_timestamps(
+        timestamp_in_days as i64 * SECONDS_IN_DAY * MILLISECONDS_IN_SECOND,
+    )
+}
+
+/// Converts an [`i64`] stored in [`arrow::array::types::Date64Type`] to PostgresSQL TimestampWithTimeZone
+///
+/// Takes into account [`arrow::array::types::Date64Type`] stores the number of milliseconds
+/// elapsed since UNIX epoch (1970-01-01).
+/// Postgres [`datum::TimestampWithTimeZone`] type takes timestamp in microseconds
+/// with epoch (2000-01-01)
+#[inline(always)]
+pub(crate) fn arrow_date64_to_postgres_timestamps(
+    timestamp_in_milliseconds: i64,
+) -> Result<Option<TimestampWithTimeZone>, FromTimeError> {
+    DateTime::from_timestamp_millis(timestamp_in_milliseconds)
```

Comment:
This is a bit complicated to read, I don't think we need to do all this custom conversion. You can just get the year/month/day from the arrow date and pass it into `TimestampWithTimeZone::new()`.

Reply:
You are right about this. There are more utilities for this already in the code base.